plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -685,6 +700,40 @@ export default createWidget("discourse-poll", {
     this.state.showResults = !this.state.showResults;
   },
 
+  exportResults() {
+    const { attrs } = this;
+    const queryID = this.siteSettings.poll_export_data_explorer_query_id;
+
+    // This uses the Data Explorer plugin export as CSV route
+    // There is detection to check if the plugin is enabled before showing the button
+    ajax(`/admin/plugins/explorer/queries/${queryID}/run.csv`, {
+      type: "POST",
+      data: {
+        params: JSON.stringify({ // needed for data-explorer route compatibility
+          poll_name: attrs.poll.name,
+          post_id: attrs.post.id.toString() // needed for data-explorer route compatibility
+        }),
+        explain: false,
+        limit: 1000000,
+        download: 1
+      }
+    }).then(csvContent => {
+      const downloadLink = document.createElement('a');
+      const blob = new Blob([csvContent],{type: 'text/csv;charset=utf-8;'});
+      const url = URL.createObjectURL(blob);
+      downloadLink.href = url;
+      downloadLink.setAttribute("download", `poll-export-${attrs.poll.name}-${attrs.post.id}.csv`);
```

Comment:
I was thinking maybe post_number would be better

Reply:
name and post_id are the primary keys for polls, so this will guarantee unique named files.

Reply:
ok all good then :+1: